### PR TITLE
lower speed test frequency and parallelism

### PR DESCRIPTION
**Pull request by willscott**
for #622

### CODE REVIEW

File: client/optimizing.go [hunk 1]
```diff
@@ -330,88 +331,200 @@ type watchResult struct {
...
 	}
+	ws.failed = nf
+}
 
-	doneChan := make(chan bool)
-	go func() {
-		wg.Wait()
-		close(doneChan)
-	}()
+func (ws *watchState) close(clientIdx int) {
```
Comment (alanshaw) [inline client/optimizing.go:471]: Take a `Client` instead of an index and make `ws.has(Client)` return a bool?

File: client/optimizing.go [hunk 2]
```diff
@@ -330,88 +331,200 @@ type watchResult struct {
...
+				ws.closeSlowest()
+				ws.tryRepopulate(resultChan, closingClients)
 			}
-			oc.updateStats([]*requestStat{&stat})
-			if round > latest {
-				latest = round
-				out <- r.Result
+		case <-ws.ctx.Done():
+			for _, c := range ws.active {
+				c.CancelFunc()
```
Comment (alanshaw) [inline client/optimizing.go:425]: Do we need to worry about canceling these and them all then trying to write to `closingClients` which we'll not read?

File: client/optimizing.go (hunk 1 above)
Comment (willscott) [reply client/optimizing.go:471]: i think that would end up needing to scan through the list of clients twice. seems nicer to only do that work once

File: client/optimizing.go (hunk 2 above)
Comment (willscott) [reply client/optimizing.go:426]: why won't we read it? added a comment to clarify

Comment (alanshaw) [reply client/optimizing.go:423]: `ws.active` is set to an empty slice of watchingClient immediately below

Comment (willscott) [reply client/optimizing.go:423]: good catch!

**Pull request merged**